clisops/utils/dataset_utils.py

```diff
@@ -207,13 +208,35 @@ def is_level(coord):
     return False
 
 
+def _is_time(coord):
+    """
+    Check if a coordinate uses cftime datetime objects.
+    Handles Dask-backed arrays for lazy evaluation.
+    """
+    if coord.size == 0:
+        return False  # Empty array
+
+    # Safely get first element without loading entire array
+    first_value = coord.isel({dim: 0 for dim in coord.dims}).values
+
+    # Compute only if it's a Dask array
+    if isinstance(first_value, da.Array):
+        first_value = first_value.compute()
+
+    return isinstance(first_value.item(), cftime.datetime)
+
+
 def is_time(coord):
     """
     Determines if a coordinate is time.
 
     :param coord: coordinate of xarray dataset e.g. coord = ds.coords[coord_id]
     :return: (bool) True if the coordinate is time.
     """
+    if coord.ndim >= 2:
```

Comment:
Should it really be `False` by default for `time_bnds`, so should `time_bnds` not be considered as `time` coordinate?